org.metaborg.core/src/main/java/org/metaborg/core/tracing/Resolution.java

```diff
@@ -15,11 +17,33 @@
     /**
      * Resolution targets. Multiple targets indicate resolution to multiple valid locations.
      */
-    public final Iterable<ISourceLocation> targets;
+    public final Iterable<Resolution.Target> targets;
 
 
-    public Resolution(ISourceRegion highlight, Iterable<ISourceLocation> targets) {
+    public Resolution(ISourceRegion highlight, Iterable<Resolution.Target> targets) {
         this.highlight = highlight;
         this.targets = targets;
     }
+
+    public static class Target {
```

Comment:
It would be better to have this as a class in a separate file (e.g., ResolutionTarget.java), instead of as a nested class, to improve code completion/import resolution in IDEs.